raysect/core/math/function/function2d/tests/test_base.py

```diff
@@ -237,4 +357,201 @@ def test_pow_3_arguments(self):
+                )
+                self.assertEqual(
+                    (ref_value != self.f1)(x, y), 0.0,
+                    msg="Callable not equals Function2D (f1() != f2()) did not return false when it should."
+                )
+                self.assertEqual(
+                    (less_value < self.f1)(x, y), 1.0,
+                    msg="Callable less than Function2D (f1() < f2()) did not return true when it should."
+                )
+                self.assertEqual(
+                    (greater_value < self.f1)(x, y), 0.0,
+                    msg="Callable less than Function2D (f1() < f2()) did not return false when it should."
+                )
+                self.assertEqual(
+                    (greater_value > self.f1)(x, y), 1.0,
+                    msg="Callable greater than Function2D (f1() > f2()) did not return true when it should."
+                )
+                self.assertEqual(
+                    (less_value > self.f1)(x, y), 0.0,
+                    msg="Callable greater than Function2D (f1() > f2()) did not return false when it should."
+                )
+                self.assertEqual(
+                    (less_value <= self.f1)(x, y), 1.0,
+                    msg="Callable less equals Function2D (f1() <= f2()) did not return true when it should."
+                )
+                self.assertEqual(
+                    (ref_value <= self.f1)(x, y), 1.0,
+                    msg="Callable less equals Function2D (f1() <= f2()) did not return true when it should."
+                )
+                self.assertEqual(
+                    (greater_value <= self.f1)(x, y), 0.0,
+                    msg="Callable less equals Function2D (f1() <= f2()) did not return false when it should."
+                )
+                self.assertEqual(
+                    (greater_value >= self.f1)(x, y), 1.0,
+                    msg="Callable equals Function2D (f1() >= f2()) did not return true when it should."
+                )
+                self.assertEqual(
+                    (ref_value >= self.f1)(x, y), 1.0,
+                    msg="Callable greater equals Function2D (f1() >= f2()) did not return true when it should."
+                )
+                self.assertEqual(
+                    (less_value >= self.f1)(x, y), 0.0,
+                    msg="Callable equals Function2D (f1() >= f2()) did not return false when it should."
+                )
+
+    def test_richcmp_function_function(self):
+        v = [-1e10, -7, -0.001, 0.0, 0.00003, 10, 2.3e49]
+        for x in v:
+            for y in v:
+                ref_value = self.f1
+                # Without Function2D.__abs__, manually ensure greater and less values are suitable
+                if self.f1(x, y) >= 0:
+                    greater_value = self.f1 + self.f1 + 1
+                    less_value = self.f1 - self.f1 - 1
+                else:
+                    greater_value = self.f1 - self.f1 + 1
```

Comment:
isn't this just 1?  Is this what was intended?

Reply:
See above

Reply:
Resolved by adding abs operator.